### DISCUSSION TITLE
Update on support for OIDC with Amazon Cognito?

### DISCUSSION THREAD

**Discussion by faermanj** [quarkusio/quarkus#40375] [Community]
Is there any guide or reference on configuring Amazon Cognito as OIDC provider for Quarkus?

**Comment by sberyozkin** [comment]
@faermanj I haven't tested it, but AFAIK, some users did. It should be configured like any other OIDC provider. If the bearer access tokens have to be checked then the address alone should be enough to discover the keys. For the code flow, the client id, secret, application type=`web-app`

**Reply by faermanj** [reply]
@AdamBien would you have an example of OIDC with Cognito? I believe you mentioned it's Cognito is non-standard in some ways but I don't recall exactly.

**Reply by sberyozkin** [reply]
@faermanj Are you seeing any specific issues with it ? 

As far as I recall, the problem was related to verifying Amazon ALB tokens, where the JWKS retrieval endpoint was dynamically detected. That already works with smallrye-jwt. However I'm not sure what is the exact relationship between Cognito and ALB.

**Reply by faermanj** [reply]
Not yet, just checking before I do something stupid :)
I will setup a small demo with only the cognito infrastructure and a "quarkus create app + oidc", let's see how it goes.
